app/components/documents/document-card.hbs

```diff
@@ -32,9 +32,7 @@
               {{#if this.loadFiles.isRunning}}
                 <AuLoader @padding="small"/>
               {{else}}
-                {{#let (or this.piece.file.derived.extension this.piece.file.extension) as |extension|}}
-                  {{this.piece.name}}.{{extension}}
-                {{/let}}
+                {{this.piece.name}}.{{this.extension}}
```

Comment:
Does the following not work?

```
{{#let (or this.piece.file.derived.extension (or this.piece.file.extension this.extension)) as |extension|}}
  {{this.piece.name}}.{{extension}}
{{/let}}
```
If this works, then `extension` doesn't need to be `@tracked` in the js, and all the `loadExtension` logic can be avoided.

Reply:
Pretty sure you can put everything in 1 `OR` statement = `(or X Y Z )`
https://github.com/jmurphyau/ember-truth-helpers